src/_pytest/python.py

```diff
@@ -125,10 +124,10 @@ def pytest_generate_tests(metafunc):
     # those alternative spellings are common - raise a specific error to alert
     # the user
     alt_spellings = ["parameterize", "parametrise", "parameterise"]
-    for attr in alt_spellings:
-        if hasattr(metafunc.function, attr):
+    for mark_name in alt_spellings:
+        if metafunc.definition.get_closest_marker(mark_name):
```

Comment:
You probably should make this check for all marks (`iter_marks`), not only the closest one.

Reply:
we just need to know if a mark exists, not if there is more than one

Reply:
Duh of course, we are already getting it by name. 👍